src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -249,24 +245,34 @@ public static function register($mode = 0)
                 };
 
                 $displayDeprecations($deprecations);
+                $isPassing = function ($mode, $deprecations) {
+                    if (DeprecationErrorHandler::MODE_WEAK === $mode) {
+                        return true;
+                    }
+                    if (DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode) {
+                        return 0 === $deprecations['unsilencedCount'] + $deprecations['remainingCount'] + $deprecations['otherCount'];
+                    }
+
+                    return 0 === $deprecations['unsilencedCount'] + $deprecations['remainingCount'] + $deprecations['remaining vendorCount'] + $deprecations['otherCount'];
+                };
 
                 // store failing status
-                $isFailing = DeprecationErrorHandler::MODE_WEAK !== $mode && $mode < $deprecations['unsilencedCount'] + $deprecations['remainingCount'] + $deprecations['otherCount'];
+                $passesBeforeShutdown = !$isPassing($mode, $deprecations);
```

Comment:
I believe the negation is incorrect: let's assume `$isPassing` returns `true` here and in the shutdown function below. `$passesBeforeShutdown` will be `false`, which causes the `if` on line 275 to evaluate to `true` and exit with a non-zero exit code.

Reply:
Oh right! Probably something I forgot to change when I decided to avoid negative logic :+1: